src/utils/validator-lookup.ts

```diff
@@ -37,7 +42,12 @@ export default function lookupValidator(validationMap: ValidatorMap): ValidatorA
     );
 
     return isPromise(validation)
-      ? (validation as Promise<ValidationResult>).then(wrapInArray)
-      : [validation];
+      ? (validation as Promise<ValidationResult>).then(result => {
+          if (Array.isArray(result) && result.length > 0) {
+            return result[0];
+          }
+          return result;
+        })
+      : validation;
```

Comment:
ref https://github.com/poteto/ember-changeset-validations/issues/228

Returns primitive; otherwise if multiple validations, returns array (note above on line 47 was followed up to ensure an array was returned; not the first one)